[iOS][screen-orientation][managed] Fix locking status bar orientation in ExpoGo

bbarthec opened the pull request:
# Why

Resolves #10485
Resolves #13450

# How

StatusBar orientation is controller by the the root view controller currently presented by the window. This statement is true for any window that is closest to the user at the time (with the highest z-index).
ExpoGo is using `AppLoadingProgressWindow` that is a separate UIWindow mounted above the application's UIWindow.
Even though this window is later on set as `hidden = true` this window's root UIViewController is still commanding the StatusBar orientation.

I'm proposing two solutions to the issue that is caused by this mechanism:
- get rid of the UIWindow in `AppLoadingProgressWindowController` altogether by `window = null` once the LoadingProgress should hide
- replace generic `UIViewController` with more clever one (manually written) that is aware of the current screen orientation settings - it would call to the kernel to get this info.

# Test Plan

- [x] `ncl#screen-orientation` examples works fine from now one (`StatusBar` was misbehaving previously)
- [x] snack from #10485 is now working correctly
- [x] example app from #13450 works as expected now

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

bbarthec committed:
[iOS][screen-orientation][managed] Fix locking status bar orientation in ExpoGo

tsapeta reviewed (commented):
Since these are new files, could you make them in Swift? 😉

bbarthec committed:
[iOS][Expo Go] Convert ViewController from obj-c to swift

As `EXAppLoadingProgressViewController` is a newly created file,
I've converted it to Swift.

lukmccall reviewed (approved):
LGTM ✅

bbarthec committed:
Apply proper Swift file formatting
